Waku package README.md documentation

Samyoul opened the pull request:
Documentation for the `waku` package. See #1944

Samyoul committed:
Added waku README.md, starting text and structure

Addresses GH-1944

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (66)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 1a8b805f | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/1/) | 2020-04-22 22:33:28 | ~55 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200422-223233-1a8b80.zip) |
| :heavy_check_mark: | 1a8b805f | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/1/) | 2020-04-22 22:38:08 | ~5 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200422-223231-1a8b80.zip) |
| :heavy_check_mark: | 1a8b805f | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/1/) | 2020-04-22 22:40:36 | ~8 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200422-223229-1a8b80.aar) |
| | | | | | | |
| :heavy_check_mark: | 5df0258b | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/2/) | 2020-04-23 12:52:17 | ~45 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200423-125131-5df025.zip) |
| :heavy_check_mark: | 5df0258b | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/2/) | 2020-04-23 12:56:59 | ~5 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200423-125129-5df025.zip) |
| :heavy_check_mark: | 5df0258b | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/2/) | 2020-04-23 12:59:38 | ~8 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200423-125126-5df025.aar) |
| | | | | | | |
| :heavy_check_mark: | 44f5209d | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/3/) | 2020-04-24 12:30:38 | ~37 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200424-123001-44f520.zip) |
| :heavy_check_mark: | 44f5209d | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/3/) | 2020-04-24 12:33:24 | ~3 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200424-122959-44f520.zip) |
| :heavy_check_mark: | 44f5209d | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/3/) | 2020-04-24 12:37:02 | ~7 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200424-122957-44f520.aar) |
| | | | | | | |
| :heavy_check_mark: | 0ac64350 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/4/) | 2020-04-24 12:43:41 | ~39 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200424-124301-0ac643.zip) |
| :heavy_check_mark: | 0ac64350 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/4/) | 2020-04-24 12:45:31 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200424-124300-0ac643.zip) |
| :heavy_check_mark: | 0ac64350 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/4/) | 2020-04-24 12:49:26 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200424-124258-0ac643.aar) |
| | | | | | | |
| :heavy_check_mark: | 33e267ff | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/5/) | 2020-04-24 12:47:26 | ~33 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200424-124652-33e267.zip) |
| :heavy_check_mark: | 33e267ff | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/5/) | 2020-04-24 12:49:18 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200424-124650-33e267.zip) |
| :heavy_check_mark: | 33e267ff | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/5/) | 2020-04-24 12:55:39 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200424-124928-33e267.aar) |
| | | | | | | |
| :heavy_check_mark: | 5feb068b | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/6/) | 2020-04-30 16:01:34 | ~54 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200430-160039-5feb06.zip) |
| :heavy_check_mark: | 5feb068b | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/6/) | 2020-04-30 16:05:59 | ~5 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200430-160038-5feb06.zip) |
| :heavy_check_mark: | 5feb068b | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/6/) | 2020-04-30 16:08:28 | ~7 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200430-160036-5feb06.aar) |
| | | | | | | |
| :heavy_check_mark: | d1f85609 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/7/) | 2020-04-30 16:07:21 | ~31 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200430-160650-d1f856.zip) |
| :heavy_check_mark: | d1f85609 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/7/) | 2020-04-30 16:09:52 | ~3 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200430-160648-d1f856.zip) |
| :heavy_check_mark: | d1f85609 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/7/) | 2020-04-30 16:14:49 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200430-160830-d1f856.aar) |
| | | | | | | |
| :heavy_check_mark: | 1072d745 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/8/) | 2020-04-30 16:12:53 | ~47 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200430-161205-1072d7.zip) |
| :heavy_check_mark: | 1072d745 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/8/) | 2020-04-30 16:14:56 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200430-161203-1072d7.zip) |
| :heavy_check_mark: | 1072d745 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/8/) | 2020-04-30 16:21:18 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200430-161452-1072d7.aar) |
| | | | | | | |
| :heavy_check_mark: | 4ef0fdfe | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/9/) | 2020-04-30 23:07:23 | ~49 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200430-230633-4ef0fd.zip) |
| :heavy_check_mark: | 4ef0fdfe | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/9/) | 2020-04-30 23:09:03 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200430-230632-4ef0fd.zip) |
| :heavy_check_mark: | 4ef0fdfe | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/9/) | 2020-04-30 23:12:17 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200430-230630-4ef0fd.aar) |
| | | | | | | |
| :heavy_check_mark: | ddea3b2c | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/10/) | 2020-05-01 13:24:56 | ~56 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200501-132359-ddea3b.zip) |
| :heavy_check_mark: | ddea3b2c | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/10/) | 2020-05-01 13:26:26 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200501-132357-ddea3b.zip) |
| :heavy_check_mark: | ddea3b2c | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/10/) | 2020-05-01 13:29:39 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200501-132356-ddea3b.aar) |
| | | | | | | |
| :heavy_check_mark: | d6514ff2 | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/11/) | 2020-05-01 14:55:01 | ~43 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200501-145418-d6514f.zip) |
| :heavy_check_mark: | d6514ff2 | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/11/) | 2020-05-01 14:56:54 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200501-145416-d6514f.zip) |
| :heavy_check_mark: | d6514ff2 | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/11/) | 2020-05-01 15:00:23 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200501-145415-d6514f.aar) |
| | | | | | | |
| :heavy_check_mark: | abdf09ff | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/12/) | 2020-05-01 15:20:53 | ~41 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200501-152012-abdf09.zip) |
| :heavy_check_mark: | abdf09ff | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/12/) | 2020-05-01 15:22:43 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200501-152010-abdf09.zip) |
| :heavy_check_mark: | abdf09ff | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/12/) | 2020-05-01 15:25:55 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200501-152008-abdf09.aar) |
| | | | | | | |
| :heavy_check_mark: | 97522df1 | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/13/) | 2020-05-01 15:50:11 | ~35 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200501-154936-97522d.zip) |
| :heavy_check_mark: | 97522df1 | [#13](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/13/) | 2020-05-01 15:51:55 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200501-154934-97522d.zip) |
| :heavy_check_mark: | 97522df1 | [#13](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/13/) | 2020-05-01 15:55:19 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200501-154932-97522d.aar) |
| | | | | | | |
| :heavy_check_mark: | 68e164d4 | [#14](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/14/) | 2020-05-01 15:51:17 | ~38 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200501-155039-68e164.zip) |
| :heavy_check_mark: | 68e164d4 | [#14](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/14/) | 2020-05-01 15:58:07 | ~6 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200501-155157-68e164.zip) |
| :heavy_check_mark: | 68e164d4 | [#14](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/14/) | 2020-05-01 16:01:05 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200501-155521-68e164.aar) |
| | | | | | | |
| :heavy_check_mark: | 17a47506 | [#15](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/15/) | 2020-05-01 16:05:27 | ~40 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200501-160446-17a475.zip) |
| :heavy_check_mark: | 17a47506 | [#15](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/15/) | 2020-05-01 16:07:32 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200501-160444-17a475.zip) |
| :heavy_check_mark: | 17a47506 | [#15](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/15/) | 2020-05-01 16:11:49 | ~7 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200501-160441-17a475.aar) |
| | | | | | | |
| :heavy_check_mark: | f615bef1 | [#16](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/16/) | 2020-05-01 16:29:58 | ~35 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200501-162923-f615be.zip) |
| :heavy_check_mark: | f615bef1 | [#16](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/16/) | 2020-05-01 16:31:59 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200501-162921-f615be.zip) |
| :heavy_check_mark: | f615bef1 | [#16](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/16/) | 2020-05-01 16:35:13 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200501-162919-f615be.aar) |
| | | | | | | |
| :heavy_check_mark: | 51ca65f9 | [#17](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/17/) | 2020-05-01 18:17:29 | ~35 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200501-181653-51ca65.zip) |
| :heavy_check_mark: | 51ca65f9 | [#17](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/17/) | 2020-05-01 18:19:24 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200501-181651-51ca65.zip) |
| :heavy_check_mark: | 51ca65f9 | [#17](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/17/) | 2020-05-01 18:22:33 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200501-181649-51ca65.aar) |
| | | | | | | |
| :heavy_check_mark: | 223f4d26 | [#18](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/18/) | 2020-05-05 23:37:02 | ~53 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200505-233609-223f4d.zip) |
| :heavy_check_mark: | 223f4d26 | [#18](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/18/) | 2020-05-05 23:39:33 | ~3 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200505-233607-223f4d.zip) |
| :heavy_check_mark: | 223f4d26 | [#18](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/18/) | 2020-05-05 23:42:18 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200505-233605-223f4d.aar) |
| | | | | | | |
| :heavy_check_mark: | 493308eb | [#19](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/19/) | 2020-05-05 23:53:46 | ~42 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200505-235304-493308.zip) |
| :heavy_check_mark: | 493308eb | [#19](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/19/) | 2020-05-05 23:55:31 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200505-235302-493308.zip) |
| :heavy_check_mark: | 493308eb | [#19](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/19/) | 2020-05-05 23:58:49 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200505-235300-493308.aar) |
| | | | | | | |
| :heavy_check_mark: | 39bc04b1 | [#20](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/20/) | 2020-05-06 00:09:46 | ~36 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200506-000909-39bc04.zip) |
| :heavy_check_mark: | 39bc04b1 | [#20](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/20/) | 2020-05-06 00:12:32 | ~3 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200506-000908-39bc04.zip) |
| :heavy_check_mark: | 39bc04b1 | [#20](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/20/) | 2020-05-06 00:14:56 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200506-000906-39bc04.aar) |
| | | | | | | |
| :heavy_check_mark: | bc17559d | [#21](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/21/) | 2020-05-06 15:36:23 | ~41 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200506-153542-bc1755.zip) |
| :heavy_check_mark: | bc17559d | [#21](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/21/) | 2020-05-06 15:41:02 | ~5 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200506-153540-bc1755.zip) |
| :heavy_check_mark: | bc17559d | [#21](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/21/) | 2020-05-06 15:41:48 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200506-153538-bc1755.aar) |
| | | | | | | |
| :heavy_check_mark: | 7c307dfc | [#22](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/22/) | 2020-05-08 17:15:57 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200508-171450-7c307d.zip) |
| :heavy_check_mark: | 7c307dfc | [#22](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/22/) | 2020-05-08 17:17:23 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200508-171448-7c307d.zip) |
| :heavy_check_mark: | 7c307dfc | [#22](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/22/) | 2020-05-08 17:21:00 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200508-171446-7c307d.aar) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 69fb55d0 | [#23](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/23/) | 2020-05-13 12:16:59 | ~48 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200513-121610-69fb55.zip) |
| :heavy_check_mark: | 69fb55d0 | [#23](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/23/) | 2020-05-13 12:18:50 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200513-121609-69fb55.zip) |
| :heavy_check_mark: | 69fb55d0 | [#23](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/23/) | 2020-05-13 12:25:42 | ~9 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200513-121557-69fb55.aar) |
| | | | | | | |
| :heavy_check_mark: | fca3f644 | [#24](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1949/24/) | 2020-05-13 12:21:46 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200513-121920-fca3f6.zip) |
| :heavy_check_mark: | fca3f644 | [#24](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1949/24/) | 2020-05-13 12:26:18 | ~6 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200513-121923-fca3f6.zip) |
| :heavy_check_mark: | fca3f644 | [#24](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1949/24/) | 2020-05-13 12:33:16 | ~7 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200513-122544-fca3f6.aar) |

Samyoul committed:
Added api, config and doc details

Samyoul committed:
Merge branch 'develop' into docs/waku

Samyoul committed:
Created Docs for waku.const, envelope and events

Samyoul committed:
Merge branch 'develop' into docs/waku

Samyoul committed:
Added notes on Filter

Samyoul committed:
Merge branch 'develop' into docs/waku

Samyoul committed:
Rewrote the filter section

Samyoul committed:
Restructured docs to match new versioning structure

Samyoul committed:
Moved package files out of the main ToC

Samyoul committed:
Added waku/common/README.md

Samyoul committed:
Added details to mailserver.go and notes to api.go

Samyoul committed:
Update to bloomfilter and const common

Samyoul committed:
Added common/ helpers.go, errors.go and message.go details

Samyoul committed:
Added common/metrics.go and protocol details

Samyoul committed:
Added rate_limiter and topic.go details

Samyoul committed:
fix rate limiter table

Samyoul committed:
Added Waku Versioning details

Samyoul committed:
Added Versioned sub-package file details

Samyoul committed:
Merge branch 'develop' into docs/waku

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [x] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status_go.html))?

cammellos commented on waku/README.md:
```diff
@@ -0,0 +1,287 @@
+Waku was [created to solve scaling issues with Whisper](https://discuss.status.im/t/fixing-whisper-for-great-profit/1419) and [currently diverges](https://github.com/vacp2p/specs/blob/master/specs/waku/waku-1.md#differences-between-shh6-and-waku1) from Whisper in the following ways:
+
+- RLPx subprotocol is changed from `shh/6` to `waku/1`.
+- Light node capability is added.
```
I believe light node was also available in whisper https://github.com/ethereum/go-ethereum/blob/510b6f90db406b697610fe0ff2eee66d173673b2/whisper/whisperv6/whisper.go#L291 , not sure if it's exactly the same, but the core concept seems to be (does not forward any envelope)

Samyoul replied:
Thanks for the feedback. Would that mean that [the specs](https://github.com/vacp2p/specs/blob/master/specs/waku/waku-1.md#differences-between-shh6-and-waku1) require clarification? As the specs say that light node capability is new.

I'll dig in the code a bit to see where the divergence is.

cammellos replied:
oh, yes, looks like the capability is not new actually, nice catch, maybe worth updating the specs if that's the case

Samyoul replied:
Raised here https://github.com/vacp2p/specs/pull/117 . If you don't mind I will wait to see what the feedback is from this PR.

Samyoul replied:
So according to the feedback I got, the Vac team feel that making this change to the specs is not desirable. I will make an amendment to the docs to reflect the discrepancy between Waku light nodes being a direct fork of Whisper light nodes and simultaneously considered a new feature.

Samyoul replied:
Addressed with https://github.com/status-im/status-go/pull/1949/commits/493308ebf152a414d66ed962d321f8d690a459ab

cammellos commented on waku/README.md:
```diff
@@ -0,0 +1,287 @@
+
+#### Usage
+
+A `status-go` node will install / register filters through RPC calls from a client (eg `status-react`). When the client installs a filter, it will specify 2 things:
```
This is a bit more complicated:
it allows to specify a `PoW` ( https://github.com/status-im/status-go/blob/4d00656c41909ccdd80a8a77a0982bd66f74d29e/waku/common/filter.go#L37 ) as well, which is the minimum `PoW` for that filter, a public key for the sender (`Src` https://github.com/status-im/status-go/blob/4d00656c41909ccdd80a8a77a0982bd66f74d29e/waku/common/filter.go#L33 )  and whether we accept `P2P` envelopes on this filter https://github.com/status-im/status-go/blob/4d00656c41909ccdd80a8a77a0982bd66f74d29e/waku/common/filter.go#L38 .
We currently don't use `Src` at all.
I believe we always set `P2P` to `true` as we always want to receive `P2P` envelopes on those filters (only from trusted peers).

And we kind of use .`PoW`:

Whisper forced you to listen to a filter if you wanted to publish to it (for light client nodes I am not sure that makes much sense, but we'd have to look into it a bit deeper).

In `status-react` each user listen to the topic `hash(pk)`, so if I want to send a message to `hash(pk1)` I will need to listen to that topic as well, but I don't really want to receive envelopes. So we listen to that topic, but we set the `PoW` to 1 so that all envelopes are discarded.

Not sure we want to document all of these (we could), but otherwise I would just change `it will specify 2 things` to `it will specify at least 2 things` :)

Samyoul replied:
It is tricky to know where the line needs to be drawn on the docs. I want to give the read enough information so that they can understand the purpose but at the same time not be so overwhelmed with information that they are confused.

I will at least change the statement that `it will specify at least 2 things` and then detail the optional fields.

Samyoul replied:
Addressed with https://github.com/status-im/status-go/pull/1949/commits/223f4d26937d41b9f7a1acb6336581f1157fb290

cammellos reviewed (approved):
Awesome work!

andremedeiros commented on waku/README.md:
```diff
@@ -0,0 +1,287 @@
+
+[`config.go`](./config.go) is home to the `Config{}` struct and the declaration of `DefaultConfig`.
+
+`Config{}` is used to initialise the settings of an instantiated `Waku{}`. `waku.New()` creates a new instance of a `Waku{}` and takes a `Config{}` as a parameter, if nil is passed instead of an instance of `Config{}`, `DefaultConfig` is used. 
```
What are the possible config values and what are they defined as with regards to defaults?

Samyoul replied:
Addressed with https://github.com/status-im/status-go/pull/1949/commits/39bc04b1958a3bc0b7161613608ee572bab8aa8a

andremedeiros commented on waku/README.md:
```diff
@@ -0,0 +1,287 @@
+
+### Version `message.go`
+
+`message.go` // TODO - I don't really understand what this does.
```
- [ ] Complete this

cammellos replied:
`message.go` is used for what we call message confirmations:

In the UI when a message is sent, it is marked initially as "Pending" and eventually as "Sent".

In order to trigger the transition from pending to "sent", there are these "message confirmations".

Each peer on receiving a message will send back one of these message responses https://github.com/status-im/status-go/blob/4d00656c41909ccdd80a8a77a0982bd66f74d29e/waku/v1/message.go#L31 .

The waku host will check that this is a mailserver and if it's been selected by the user.
If so the message will be marked as "Sent" in the UI.

https://github.com/status-im/specs/blob/master/docs/stable/3-whisper-usage.md has a section on it.

As a side not, we should not be doing any versioning at this level any more (the struct has a `Version` field), something to remove once we decide to move to `waku/2` for example.

Samyoul replied:
Resolved with https://github.com/status-im/status-go/pull/1949/commits/bc17559d4284f4d3bbdee51d6dc9dd7a63bacf7e

Samyoul replied:
@cammellos @andremedeiros 

**I'm going to implement the following:**

- Move send bundle to Peer
- change v*/message.go to v*/message_response.go - I know that I originally merged message response functionality into message, but after the `waku/1` refactor it makes sense to move it back.

**I've got some questions:**

- Should `MultiVersionResponse` and `Version1MessageResponse` be merged into a single struct MessageResponse? Or at least renamed to be more obvious what they are.
- Should the structs in v*/message be unexported as they are only consumed inside the version subpackage?
- Should Waku "message response" be renamed as "message confirmation"? It is called a message confirmation in the docs, but also a message response, it makes more sense as confirmation because a message response can be anything that responds to a message including a confirmation, a confirmation is explicitly a confirmation.
- Should MessagesResponse be renamed to MessageResponse
- Should we track new features for the next Waku version? If so where would be a good place?

See :
>Versioning at the MessageResponse level (see the struct Version field) should be phased out and defer to the subpackage's version number. Consider removal once we decide to move to a new major Waku version (i.e. waku/2).

cammellos replied:
> Should MultiVersionResponse and Version1MessageResponse be merged into a single struct MessageResponse? Or at least renamed to be more obvious what they are.

I would say so, if feasible one single struct makes sense.

> Should the structs in v*/message be unexported as they are only consumed inside the version subpackage?

Yes

> Should Waku "message response" be renamed as "message confirmation"?

I'd think so, not sure we use it for anything other than messages confirmations

> Should we track new features for the next Waku version? If so where would be a good place?

I'd think that would be the specs repo, as any new feature should be driven by the specs I take, what do you think?

Samyoul replied:
I agree. I was thinking about making the renaming and merging changes now in the open tidy PR, but on reflection I think that we need consensus on the use of terminology and the direction we want to take improvements to the specs.

I will open a spec issue on this subject once I'm done [updating the status specs](https://github.com/status-im/specs/issues/61).

jakubgs commented on waku/README.md:
```diff
@@ -0,0 +1,287 @@
+- Light node capability is added.
+- Optional rate limiting is added.
+- Status packet has following additional parameters: light-node, confirmations-enabled and rate-limits
+- Mail Server and Mail Client functionality is now part of the specification.
```
Worst naming ever, but I guess I will never be able to push through the renaming of this to `history node`.

Samyoul replied:
He who dares wins. I support this change and would lobby for its implementation.

If we can agree on the better name. I would propose `persistence node`, or `echo node`.

One thing I am 100% in agreement with, the Mailserver is a stupid name.

jakubgs replied:
Yeah, makes you think of technology from 1970s, not something developed in 2020s.

Samyoul replied:
@jakubgs this may interest you https://discuss.status.im/t/wherefore-art-thou-mailserver-treatise-on-waku-terminology/1664

Samyoul committed:
Added better detail for Waku Filters

Addressing https://github.com/status-im/status-go/pull/1949#discussion_r419232235

Samyoul committed:
Added notes about light nodes not being actually new

Samyoul committed:
Added more detail for Waku node configs and defaults

Samyoul committed:
Added details to the version message.go section

Samyoul committed:
Merge branch 'develop' into docs/waku

Samyoul committed:
Added details to waku/common/rate_limiter

Samyoul committed:
Minor typo fix in docs

Samyoul commented:
@cammellos, @jakubgs, @andremedeiros Would you mind giving this PR a final review. I believe it is ready for merging, but I'd like the new updates to be checked.

cammellos commented:
looks good, thanks!
